src/component/handlers/edit/editOnBeforeInput.js

```diff
@@ -173,11 +173,25 @@ function editOnBeforeInput(editor: DraftEditor, e: SyntheticInputEvent): void {
     editor._waitingOnInput = true;
```

Comment:
I believe we can move the following two lines behind the `if (finalNativelyRenderedContent...` check. That way we don't have to explicitly set them to false below.